rust/lancedb/src/table.rs

```diff
@@ -391,7 +390,12 @@ pub(crate) trait TableInternal: std::fmt::Display + std::fmt::Debug + Send + Syn
         query: &Query,
         options: QueryExecutionOptions,
     ) -> Result<DatasetRecordBatchStream>;
-    async fn explain_plan(&self, query: &VectorQuery, verbose: bool) -> Result<String>;
+    async fn explain_plan(&self, query: &VectorQuery, verbose: bool) -> Result<String> {
+        let plan = self.create_plan(query, Default::default()).await?;
+        let display = DisplayableExecutionPlan::new(plan.as_ref());
+
+        Ok(format!("{}", display.indent(verbose)))
+    }
```

Comment:
I suppose this works for remote table but it will always just show `OneShotExec` right?

Reply:
It does. I was thinking maybe one day we will have a `RemoteQueryExec` that can handle federation or something. Like maybe do a client side join could be chained on after.